.github/workflows/ci.yml

```diff
@@ -17,17 +17,47 @@ jobs:
           -G "Unix Makefiles" \
           -DCMAKE_BUILD_TYPE=Debug \
           -DBUILD_CLONE_SUBMODULES=ON \
-          -DCMAKE_C_FLAGS='-Wall -Wextra -Werror'
+          -DCMAKE_C_FLAGS='-O0 -Wall -Wextra -Werror -Wformat -Wformat-security -Warray-bounds'
           make -C build/ coverity_analysis -j8
       - name: Build Library in Release mode
         run: |
           rm -rf ./build
+          # Build with logging enabled.
           cmake -S test -B build/ -G "Unix Makefiles" \
           -DCMAKE_BUILD_TYPE=Release \
           -DBUILD_CLONE_SUBMODULES=ON \
-          -DCMAKE_C_FLAGS='-Wall -Wextra -Werror -DNDEBUG'
+          -DCMAKE_C_FLAGS='-Wall -Wextra -Werror -DNDEBUG -Wno-error=pedantic -Wno-variadic-macros -DLOGGING_LEVEL_DEBUG=1 -Wformat -Wformat-security -Warray-bounds'
```

Comment:
I would add -D_FORTIFY_SOURCE=2 or -D_FORTIFY_SOURCE=1 here as well since some of the FORTIFY_SOURCE=1 checks are compile time rather than runtime.

Reply:
Added `-D_FORTIFY_SOURCE=1`